packages/context/src/binding.ts

```diff
@@ -3,16 +3,21 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
+import {Context, Constructor} from './context';
+
 // tslint:disable-next-line:no-any
-type BoundValue = any;
+export type BoundValue = any;
 
 export class Binding {
   // FIXME(bajtos) The binding class should be parameterized by the value type stored
   public value: BoundValue;
   public getValue: () => BoundValue = () => { throw new Error(`No value was configured for binding ${this._key}.`); };
   private _tagName: string;
 
-  constructor(private readonly _key: string, public isLocked: boolean = false) {}
+  // For bindings bound via toClass, this property contains the constructor function
+  public valueConstructor: Constructor<BoundValue>;
```

Comment:
I need to access the constructor in order to retrieve `@api` metadata. I am not very happy with `.valueConstructor` property though, I think the main reason is because the property is used only by `toClass` bindings.

Any suggestions on how to improve this design?